README.md

```diff
@@ -7,37 +7,58 @@ The **T**-online **API** **R**epository contains the interface definitions of t-
 that support ~~both REST and~~ gRPC protocols. You can use these definitions with open source
 tools to generate client libraries, documentation and other artifacts.
 
-T-online APIs use [Protocol Buffers](https://github.com/google/protobuf)
-version 3 (proto3) as their Interface Definition Language (IDL) to
-define the API interface and the structure of the payload messages.
+T-online APIs use [Protocol Buffers](https://github.com/google/protobuf) version 3 (proto3) 
+as their Interface Definition Language (IDL) to define the API interface and the structure of the 
+payload messages.
 
-## Building
+## building
 
-We currently support:
+we currently support generating and packaging code for `java` and `node/TypeScript`.
 
-- Java
-- Node (incl. typescript types)
+### java
 
-## Installation and Development
+#### generate
+ 
+We use [docker-protobuf](https://github.com/TheThingsIndustries/docker-protobuf) as `protoc`. For generating
+Java sources run:
 
 ```bash
-# protoc
-$ brew install protobuf
+$ make
+```
+
+#### release
+
+The recommended way to build and publish the Java client libraries is through gradle:
+
+```shell script
+$ export GITHUB_USERNAME=foo
+$ export GITHUB_TOKEN=bar
+$ cd java && ./gradlew publish
+``` 
+
+### node
 
-# protoc-gen-grpc
-# ts-protoc-gen
+#### generate
+
+Since [docker-protobuf](https://github.com/TheThingsIndustries/docker-protobuf) doesn't support TypeScript 
```

Comment:
Warum bauen wir dann nicht einfach unser eigenes docker image? Dafür ist docker doch da ?

Reply:
we could do that - or fix the issue in TheThingsIndustries/docker-protobuf